### ISSUE TITLE
Keepassxc-proxy.exe continually respawns/crashes when keepassxc isn't running, blocking update install

### ISSUE THREAD

**Issue by Tatsu-Kishi** [keepassxreboot/keepassxc#7836]
## Overview
Just updated to 2.7.1 from 2.7.0 which got held up by keepassxc-proxy immediately respawning upon getting killed. Additionally when keepassxc itself is not running and the proxy is killed it continually crashes/respawns.
For the update itself I was able to circumvent the issue by starting keepassxc when the installer was asking to kill the remaining proxy process and then continuing the install.

## Steps to Reproduce
1. Close keepassxc
2. Have browser with plugin open
3. kill keepassxc-proxy.exe in task manager
4. watch continual respawns/crashes

## Expected Behavior
keepassxc-proxy doesn't respawn when keepassxc isn't running. Especially doesn't crashloop. Especially especially during an update install.

## Actual Behavior
keepassxc-proxy respawns/crashloops. Sometimes even when just quitting keepassxc (observed once during testing).

## Context
KeePassXC - 2.7.1 (also observed in 2.7.0)
Revision: 5916a8f
Operating System: Windows 10

## Comment
The behaviour could almost be called malicious - a process whose main-process (keepassxc.exe) isn't running shouldn't respawn by itself.

For uninstalls and updates at least a workaround might be to create a lock file that prevents the plugins from starting the proxy while it exists. But in the longterm the proxy should be entirely controlled by keepassxc (again), not the plugins.

**Comment by droidmonkey** [comment]
The browser spawns the proxy as part of native messaging.  What browser are you using? The status of keepassxc.exe is not relevant to the browser spawning the proxy. 

When you do quit keepassxc we have programmed in for the proxy to close, but the extension/browser may start it again as needed or requested. 

Quitting or closing does not mean it is crashing. That term is specific to a software fault that causes the application to be killed by the kernel.

**Comment by Tatsu-Kishi** [comment]
I'm using both Firefox and Chrome simultaneously (both with the extension installed). The reason I'm claiming crashing is because it constantly changes PID in taskmanager which I'm assuming isn't what's supposed to happen. That stops once keepassxc is running again.
The more apt term is probably closing unexpectedly - at least as viewed by the extensions.

It seems weird for the extension to initiate the starting of the proxy that from my understanding exists as part of the server-component of keepassxc. Of course the extension wants to continually connect to keepassxc, but when keepassxc isn't running it seems pointless to let it go as far as starting the proxy for the proxy to then notice that there's nothing to connect to.

**Comment by droidmonkey** [comment]
The proxy is a client, keepassxc is the server. PID change normal whenever a process is closed and started again. Do you have "automatically connect to keepassxc" enabled in the extension settings. 

Recommend reading up on how native messaging extensions work for more details on why this stuff happens.

**Comment by Tatsu-Kishi** [comment]
Firefox had autoconnect active - I deactivated it for now (I also don't think I activated it myself... haven't been in the settings for ages). But the autoconnect setting basically preventing updates is still far from ideal.

**Comment by droidmonkey** [comment]
It's disabled by default, and I agree this has been a problem. It's also a problem on macos we found.

**Comment by Xeevis** [comment]
Found this to be issue as well. When I looked in the task manager to end the conflicting process it's gonna take a long time to end ...
I'm using Vivaldi and `Automatically reconnect to KeePassXC` enabled. (There are twice as many processes than screenshot shows) isn't it supposed to be just one instance?

![image](https://user-images.githubusercontent.com/5835044/162586803-32620a5e-bab0-4b9b-b779-082cae987f79.png)
